crates/ide-completion/src/item.rs

```diff
@@ -205,6 +207,26 @@ pub enum CompletionRelevancePostfixMatch {
     Exact,
 }
 
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub struct CompletionRelevanceFn {
+    pub has_args: bool,
+    pub has_self_arg: bool,
+    pub ty: CompletionRelevanceFnType,
+}
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum CompletionRelevanceFnType {
+    Other,
+    /// Returns a type that is expected in this context
+    ReturnsExpectedType,
```

Comment:
This variant should not be needed, iirc we set `type_match` according to the return type for functions when needed.